### PR TITLE
Add multi tag based feeds for HTML and RSS views

### PR DESCRIPTION
**Pull request by sohaibbhatti**
https://github.com/lobsters/lobsters/issues/525

Implements multi tag based feeds. Identified a possible refactoring. Please have a look at the code snippet in the comments. Essentially, using a JOIN over a nested query. Thoughts?

### CODE REVIEW

**Comment by sohaibbhatti** [comment]
Odd. Tests are passing locally.

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -187,19 +187,23 @@ def saved
...
 
   def tagged
-    @tag = Tag.where(:tag => params[:tag]).first!
+    tag_params = params[:tag].split('+')
+    @tags = Tag.where(tag: tag_params)
+    @tags.first!
 
-    @stories, @show_more = get_from_cache(tag: @tag) {
-      paginate stories.tagged(@tag)
+    @stories, @show_more = get_from_cache(tags: tag_params.sort.join('+')) {
```
[the remark this answers is not in the document]
Comment (pushcx) [reply app/controllers/home_controller.rb]: Yeah, good touch.

File: app/models/story_repository.rb [hunk 2]
```diff
@@ -29,11 +29,11 @@ def saved
     saved.order("hotness")
   end
 
-  def tagged(tag)
+  def tagged(tags)
     tagged = Story.base.positive_ranked.where(
```
[the remark this answers is not in the document]
Comment (pushcx) [reply app/models/story_repository.rb]: Yeah, if we can express this with the regular AR query interface without digging down to Arel that's a plus.

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -299,4 +303,8 @@ def get_from_cache(opts = {}, &block)
   def user_token_link(url)
     @user ? "#{url}?token=#{@user.rss_token}" : url
   end
+
+  def tags_with_description_for_rss(tags)
+    tags.map {|tag| "#{tag.tag} (#{tag.description})" }.join(' ')
```
Comment (pushcx) [inline app/controllers/home_controller.rb]: Nitpick: please `.join(', ')`, it'll read a little better.

**Comment by pushcx** [comment]
Looks like Travis was just having a bad day on the first build. I restarted and it passed.

Looks like there's just small tweaks (the arel -> AR and comma nitpick) before this is ready to merge.

Thanks for taking this on with a clear and clean implementation.

**Comment by sohaibbhatti** [comment]
Thanks for the review @pushcx 
Will have the changes ready tomorrow

File: app/controllers/home_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rdivyanshu) [reply app/controllers/home_controller.rb]: c++ is one of tag [https://lobste.rs/tags]. Calling split on params[:tag] will diminish into c tag.

**Comment by pushcx** [comment]
@sohaibbhatti I'm taking a pass through all the open PRs, how goes? If you don't think you'll be able to pick up these changes, I could probably help finish this code.

**Pull request closed without merging**